File: cinema-customer-frontend/src/app/user/dtos/response/ticket.response.ts
import {SeatResponse} from "../../../repertory/dtos/response/seat.response";
import {ScreeningResponse} from "../../../repertory/dtos/response/screening.response";

export class TicketResponse {
  constructor(
    readonly qrCode: string,
    readonly bookedSeats: SeatResponse[],
    readonly screening: ScreeningResponse,
    readonly cinemaName: string
  ) {
  }
}

File: cinema-backend/src/main/java/org/example/cinemabackend/ticketing/application/dto/response/TicketResponse.java
package org.example.cinemabackend.ticketing.application.dto.response;

import lombok.Builder;
import org.example.cinemabackend.cinema.application.dto.response.ScreeningResponse;
import org.example.cinemabackend.cinema.application.dto.response.SeatResponse;

import java.util.List;

@Builder
public record TicketResponse(
        String qrCode,
        List<SeatResponse> bookedSeats,
        ScreeningResponse screening,
        String cinemaName
) {
}
